### PR TITLE
feat(fuse): add prewhere support in native storage format

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

1. add prewhere support in native storage format
2. add clustering page index in native storage format
3. add page filter, this improves hits Q37 50% and other queries have no bad effects.
4. bump opendal & strawboat

```
-- before
MySQL [(none)]> select  *  from hits_native where URL like '%google%' ignore_result;
Empty set (17.149 sec)


-- after
MySQL [(none)]> select  *  from hits_native where URL like '%google%' ignore_result;
Empty set (10.402 sec)
```

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
Will the `prewhere` also works for the fuse engine with parquet format?

**Comment by sundy-li** [comment]
> Will the `prewhere` also works for the fuse engine with parquet format?

No, just work on the native format.

**Comment by Xuanwo** [comment]
> -- before
> MySQL [(none)]> select  *  from hits_native where URL like '%google%' ignore_result;
> Empty set (17.149 sec)
> 
> 
> -- after
> MySQL [(none)]> select  *  from hits_native where URL like '%google%' ignore_result;
> Empty set (10.402 sec)

This bench is on local fs?

**Comment by Xuanwo** [comment]
@sundy-li, opendal has released v0.25, can you bump to tagged version instead? Thanks.

**Comment by sundy-li** [comment]
> This bench is on local fs?

Yes，I'm focus on improving native format now.

**Comment by sundy-li** [comment]
> > This bench is on local fs?
> 
> Yes，I'm focus on improving native format now.  But I think it can also take effect on s3.

File: src/query/service/src/pipelines/pipeline_builder.rs [hunk 1]
```diff
@@ -345,7 +345,10 @@ impl PipelineBuilder {
             &aggregate.agg_funcs,
         )?;
 
-        if self.ctx.get_cluster().is_empty()
+        // this has bugs now, so we disable it for now, cc @winter
```
Comment (BohuTANG) [inline src/query/service/src/pipelines/pipeline_builder.rs:348]: File a new open issue for it?

File: src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs [hunk 2]
```diff
@@ -40,31 +52,103 @@ pub struct NativeDeserializeDataTransform {
...
+    prewhere_filter: Arc<Option<Expr>>,
+
+    prewhere_skipped: usize,
 }
 
 impl NativeDeserializeDataTransform {
     pub fn create(
         ctx: Arc<dyn TableContext>,
         block_reader: Arc<BlockReader>,
+        plan: &DataSourcePlan,
```
Comment (BohuTANG) [inline src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs:70]: Can we remove the `plan` to use `pushdwon`?
We have tried to decouple the planner with storage.

Comment (sundy-li) [reply src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs:70]: we also need the plan.source_info in `NativeDeserializeDataTransform`

File: src/query/service/src/pipelines/pipeline_builder.rs (hunk 1 above)
Comment (sundy-li) [reply src/query/service/src/pipelines/pipeline_builder.rs:348]: https://github.com/datafuselabs/databend/issues/9673

File: src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs (hunk 2 above)
Comment (BohuTANG) [reply src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs:70]: `plan.source_info.schema()` --> `schema`

Comment (BohuTANG) [reply src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs:70]: We have done much work to decouple `planners` for many crates in the past.

Comment (BohuTANG) [reply src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs:70]: Oh, i see.
This is common_catalog, not planners.

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**